tests/cnf/core/network/sriov/tests/qinq.go

```diff
@@ -544,6 +578,152 @@ var _ = Describe("QinQ", Ordered, Label(tsparams.LabelQinQTestCases), ContinueOn
+			}
+
+			By("Creating SR-IOV VFs via NMState")
+			err = netnmstate.ConfigureVFsAndWaitUntilItsConfigured(
+				configureNMStatePolicyName,
+				sriovInterfacesUnderTest[0],
+				NetConfig.WorkerLabelMap,
+				5,
+				netparam.DefaultTimeout)
+			Expect(err).ToNot(HaveOccurred(), "Failed to create VFs via NMState")
+
+			err = sriovenv.WaitUntilVfsCreated(workerNodeList, sriovInterfacesUnderTest[0], 5, netparam.DefaultTimeout)
+			Expect(err).ToNot(HaveOccurred(), "Expected number of VFs are not created")
+
+			By("Configure SR-IOV with flag ExternallyManaged true")
+			err = sriovenv.CreateSriovNmStateConfiguration(sriovAndResourceNameExManagedTrue, sriovInterfacesUnderTest[0], true)
+			Expect(err).ToNot(HaveOccurred(),
+				fmt.Sprintf("Failed to create sriov configuration via nmstate %s", err))
+
+			By("Define and create sriov-network for the promiscuous client")
+			_, err = sriov.NewNetworkBuilder(APIClient,
+				srIovNetworkPromiscuous, NetConfig.SriovOperatorNamespace, tsparams.TestNamespaceName,
+				sriovAndResourceNameExManagedTrue).WithTrustFlag(true).Create()
+			Expect(err).ToNot(HaveOccurred(),
+				fmt.Sprintf("Failed to create sriov network srIovNetworkPromiscuous %s", err))
+
+			By("Define and create sriov-network with 802.1ad S-VLAN")
+			vlan, err := strconv.Atoi(NetConfig.VLAN)
+			Expect(err).ToNot(HaveOccurred(), "Failed to convert VLAN value")
+			defineAndCreateSrIovNetworkWithQinQ(srIovNetworkDot1AD, sriovAndResourceNameExManagedTrue, dot1ad,
+				uint16(vlan))
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to create sriov network srIovNetworkDot1AD %s", err))
+			By("Verify SR-IOV Device IDs for interface under test")
+			sriovDeviceID = discoverInterfaceUnderTestDeviceID(srIovInterfacesUnderTest[0],
+				workerNodeList[0].Definition.Name)
+			Expect(sriovDeviceID).ToNot(BeEmpty(), "Expected sriovDeviceID not to be empty")
+
+			promiscVFCommand = fmt.Sprintf("ethtool --set-priv-flags %s vf-true-promisc-support on",
+				srIovInterfacesUnderTest[0])
+			if sriovDeviceID == netparam.MlxDeviceID {
+				promiscVFCommand = fmt.Sprintf("ip link set %s promisc on",
+					srIovInterfacesUnderTest[0])
+			}
+
+			By(fmt.Sprintf("Enable VF promiscuous support on %s", srIovInterfacesUnderTest[0]))
+			output, err := cmd.RunCommandOnHostNetworkPod(workerNodeList[0].Definition.Name,
+				NetConfig.SriovOperatorNamespace, promiscVFCommand)
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to run command on node %s", output))
+		})
+
+		It("Verify an 802.1ad QinQ tunneling between two containers with the VFs configured by NMState",
+			reportxml.ID("71681"), func() {
+				By("Define and create sriov-network with 802.1ad S-VLAN")
+				vlan, err := strconv.Atoi(NetConfig.VLAN)
+				Expect(err).ToNot(HaveOccurred(), "Failed to convert VLAN value")
+				defineAndCreateSrIovNetworkWithQinQ(srIovNetworkDot1AD, sriovAndResourceNameExManagedTrue, dot1ad,
```

Comment:
But you did it in beforeall, right? line 640

Reply:
Do you mean in BeforeAll in the beginning of the test?
These sriovnetworks are with sriovAndResourceNameExManagedTrue